### PR TITLE
Refactor the ordering function to prevent unnecessary dict access

### PR DESCRIPTION
**Pull request by eendebakpt**
#### Brief description of what is fixed or changed

Refactor the `ordering` function to prevent unnecessary `dict` access

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/sorting.py [hunk 1]
```diff
@@ -283,9 +283,11 @@ def ordered(seq, keys=None, default=True, warn=False):
...
     if keys:
         if not isinstance(keys, (list, tuple)):
-            keys = [keys]
-        keys = list(keys)
-        f = keys.pop(0)
+            f = keys
+            keys = []
+        else:
+            keys = list(keys)
+            f = keys.pop(0)
```
Comment (eagleoflqj) [inline sympy/core/sorting.py:290]: Could you please swap the if and else, as there is no need for using `not`?

**Comment by eendebakpt** [comment]
I think so as well. The only reason I kept it like this, is that the pop
can release some memory. I will update the pr with your suggestion

Op ma 6 jun. 2022 19:01 schreef Liumeo ***@***.***>:

**Comment by eendebakpt** [comment]
@eagleoflqj Thanks for the review, I addressed the comments.

**Review by eagleoflqj — APPROVED** [review]
LGTM. Squashing and merging.

**Pull request merged**